src/Linter.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace SLLH\ComposerLint;
+
+/**
+ * @author Sullivan Senechal <soullivaneuh@gmail.com>
+ */
+final class Linter
+{
+    /**
+     * @param array $manifest composer.json file manifest
+     *
+     * @return string[]
+     */
+    public function validate($manifest)
+    {
+        $errors = array();
+
+        if (isset($manifest['config']['sort-packages']) && $manifest['config']['sort-packages']) {
+            foreach (array('require', 'require-dev', 'conflict', 'replace', 'provide', 'suggest') as $linksSection) {
+                if (array_key_exists($linksSection, $manifest) && !$this->packagesAreSorted($manifest[$linksSection])) {
+                    array_push($errors, 'Links under '.$linksSection.' section are not sorted.');
+                }
+            }
+        }
+
+        return $errors;
+    }
+
+    private function packagesAreSorted(array $packages)
+    {
+        $names = array_keys($packages);
+
+        $hasPHP = in_array('php', $names);
+        $extNames = array_filter($names, function ($name) {
+            return 'ext-' === substr($name, 0, 4) && !strstr($name, '/');
```

Comment:
> > Agree or not, SF coding standard includes Yoda condition, so we will follow it

> I don't remember voting for that.

http://symfony.com/doc/current/contributing/code/standards.html#structure

> Use Yoda conditions when checking a variable against an expression to avoid an accidental assignment inside the condition statement (this applies to `==`, `!=`, `===`, and `!==`);

This is it from a long time and I already point this out on some sonata PRs.

Trust me, I was not a big fan of Yoda condition either. But if we start not following standards just because someone don't like long line class declaration or someone else like Star Wars or not, the standards will be useless.